author/job.py

```diff
@@ -244,6 +248,17 @@ def submit(self, dump_job=True, **kwargs):
             _LOG.info("Option to inherit environment was enabled. Passing the environment to the job.")
             self.envkey = [self._get_env_as_tractor_envkey()]
 
+        if expandchunk:
+            # construct a temporary .alf file
+            _tmp_directory = tempfile.gettempdir()
+            alf_file = os.path.join(_tmp_directory, "{}.alf".format(str(uuid.uuid4())))
+            self.dump_job(alf_file)
+
+            # and expand it
+            print("TR_EXPAND_CHUNK \"{}\"".format(alf_file))
```

Comment:
A normal print is not guaranteed to be enough in this case. We have to make sure to flush `stdout`.

Can't we reuse existing `__EXPAND__` functionality to invoke the expansion?

Reply:
> Can't we reuse existing `__EXPAND__` functionality to invoke the expansion?

I was trying, but quickly noticed it would be much more ugly. The `__EXPAND__` only exists on the `TASK`  and not the job and its not supporting passing through a Task instance yet; only a `str` or `TaskWithOverrides`.
`Task.__EXPANDS__()` in general lacks some other important features and we need to rework this soon. So far it is not really adopted so I didn't consider following this route.

Reply:
After a quick conversation it seems like we're using this mechanism everywhere so it seems safe enough.